api/src/main/java/edu/cornell/mannlib/vitro/webapp/beans/CaptchaServiceBean.java

```diff
@@ -0,0 +1,265 @@
+            .addBorder();
+
+        if (difficulty.equals("hard")) {
+            imageCaptchaBuilder
+                .addNoise(new CurvedLineNoiseProducer(getRandomColor(), 2f))
+                .addFilter(new StretchImageFilter())
+                .addFilter(new FishEyeImageFilter())
+                .build();
+        }
+
+        ImageCaptcha imageCaptcha = imageCaptchaBuilder.build();
+        return new CaptchaBundle(convertToBase64(imageCaptcha.getImage()), imageCaptcha.getContent(),
+            UUID.randomUUID().toString());
+    }
+
+    /**
+     * Retrieves a CAPTCHA challenge for a specific host based on the provided CAPTCHA ID
+     * Removes the challenge from the storage after retrieval.
+     *
+     * @param captchaId The CAPTCHA ID to match.
+     * @return An Optional containing the CaptchaBundle if a matching challenge is found,
+     * or an empty Optional otherwise.
+     */
+    public static Optional<CaptchaBundle> getChallenge(String captchaId) {
+        CaptchaBundle challengeForHost = captchaChallenges.getIfPresent(captchaId);
+        if (challengeForHost == null) {
+            return Optional.empty();
+        }
+
+        captchaChallenges.invalidate(captchaId);
+
+        return Optional.of(challengeForHost);
+    }
+
+    /**
+     * Gets the map containing CAPTCHA challenges for different hosts.
+     *
+     * @return A ConcurrentHashMap with host addresses as keys and CaptchaBundle objects as values.
+     */
+    public static Cache<String, CaptchaBundle> getCaptchaChallenges() {
+        return captchaChallenges;
+    }
+
+    /**
+     * Converts a BufferedImage object to Base64 format.
+     *
+     * @param image The BufferedImage to convert.
+     * @return The Base64-encoded string representation of the image.
+     * @throws IOException If an error occurs during image conversion.
+     */
+    private static String convertToBase64(BufferedImage image) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(image, "png", baos);
+        byte[] imageBytes = baos.toByteArray();
+
+        return Base64.getEncoder().encodeToString(imageBytes);
+    }
+
+    /**
+     * Retrieves the configured captcha implementation based on the application's configuration properties.
+     * If captcha functionality is disabled, returns "NONE." If the captcha implementation is not specified,
+     * defaults to "NANOCAPTCHA."
+     *
+     * @return The selected captcha implementation ("NANOCAPTCHA," "RECAPTCHA," or "NONE").
+     */
+    public static String getCaptchaImpl() {
+        if (!Boolean.parseBoolean(ConfigurationProperties.getInstance().getProperty("captcha.enabled"))) {
+            return "NONE";
+        }
+
+        String captchaImpl =
+            ConfigurationProperties.getInstance().getProperty("captcha.implementation");
+        if (captchaImpl == null) {
+            captchaImpl = "NANOCAPTCHA";
```

Comment:
Dragan wrote that values should be case insensitive, so this method is a good place to do case insensitive equality check.
I would refactor implementation options to an enumeration and return one of enumeration options here.